### PR TITLE
Add StorageDurationCurveRecoder.

### PR DESCRIPTION
**Pull request by jetuk**

### CODE REVIEW

File: tests/models/timeseries3.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+    "metadata": {
+        "title": "Timeseries example",
+        "description": "A model with a timeseries",
+        "minimum_version": "0.1"
```
Comment (snorfalorpagus) [inline tests/models/timeseries3.json]: Update this

File: pywr/recorders/_recorders.pyx [hunk 2]
```diff
@@ -573,6 +573,80 @@ cdef class NumpyArrayStorageRecorder(StorageRecorder):
...
+        """ Return a `pandas.DataFrame` of the recorder data
+
+        This DataFrame contains a MultiIndex for the columns with the recorder name
+        as the first level and scenario combination names as the second level. This
+        allows for easy combination with multiple recorder's DataFrames
+        """
+        index = self._percentiles
+        sc_index = self.model.scenarios.multiindex
+
+        return pd.DataFrame(data=np.array(self.sdc), index=index, columns=sc_index)
```
Comment (snorfalorpagus) [inline pywr/recorders/_recorders.pyx]: The call to `np.array` is redundant.

File: tests/test_recorders.py [hunk 3]
```diff
@@ -130,6 +131,37 @@ def test_fdc_dev_recorder():
     df = rec.to_dataframe()
     assert df.shape == (len(percentiles), len(model.scenarios.combinations))
 
+
+def test_sdc_recorder():
+    """
+    Test the StorageDurationCurveRecorder
+    """
+    model = load_model("timeseries3.json")
+    input = model.nodes['catchment1']
```
Comment (snorfalorpagus) [inline tests/test_recorders.py]: Don't overload builtins.

**Comment by jetuk** [comment]
Comments addressed.

**Pull request merged**